src/serialization/sb2.js

```diff
@@ -133,17 +133,20 @@ const parseBlockList = function (blockList, addBroadcastMsg, getVariableId, exte
         // eslint-disable-next-line no-use-before-define
         const parsedBlockAndComments = parseBlock(block, addBroadcastMsg, getVariableId,
             extensions, comments, commentIndex);
-        const parsedBlock = parsedBlockAndComments[0];
-        // Update commentIndex
-        commentIndex = parsedBlockAndComments[1];
 
-        if (typeof parsedBlock === 'undefined') continue;
-        if (previousBlock) {
-            parsedBlock.parent = previousBlock.id;
-            previousBlock.next = parsedBlock.id;
+        if (parsedBlockAndComments) { // Could fail due to unrecognized op-codes
```

Comment:
If a block that is undefined in Scratch 3.0 had a block comment on it in Scratch 2.0, it should probably get removed from the comments list. Right now, it gets attached to the next block in the stack (if there are more blocks in the stack) or doesn't display anywhere if there are no more blocks in the stack, but there's still a comment associated with it (that isn't done being parsed -- e.g. a blockId gets associated with it), so saving and loading that project in 3.0 then fails.

Reply:
So that actually means two things: 
1) `commentIndex` should still be updated in `parseBlock` even if it returns undefined (because that commentIndex refers to a position in a 2.0 stack).
2) Any block comment associated with an undefined block should get removed from the target's comments map (so that it doesn't get serialized later)

Testing this made me realized that this is also an actual issue for how I'm handling importing comments from 2.0 where multiple block comments are attached to the same block. I'll file a separate issue for that.